### DISCUSSION TITLE
Image button?

### DISCUSSION THREAD

**Discussion by preetamsad** [JakeStanger/ironbar#205] [Q&A] [answered]
Is there a way to add an image within a button?? I don't know if it is possible by nesting the image and button widgets, if it is I can't seem to get it work. Please help!!

**Comment by JakeStanger** [comment, marked as answer]
Hey, I assume this is for the `custom` module? You could use a `Box` and use the module-level `on_click_left` config option (you can add these onto any custom widget), but there is currently no way of nesting elements inside buttons. 

If you'd like this, feel free to open an issue.

**Reply by preetamsad** [reply]
Yes for the `custom` module. Oh, i totally forgot about the module-level options yes. While this will work, but then i will loose the ability to use popups because  i don't think it is possible to trigger a bar command like `popup:toggle` from anywhere but a `button`  and a `slider` widget, right?

And it is a common practice to add image/icons to buttons, so it would be cool if we can do that too.

If you agree i will open up an issue.

**Reply by JakeStanger** [reply]
Yeah, you're limited to scripts for those options at the moment, so cannot run the bar commands. I agree it'd definitely be useful to have.

**Comment by yavko** [comment]
Just wanted to say you can try css and the background property for now
